service/history/historyEngine.go

```diff
@@ -2031,11 +2031,11 @@ func (e *historyEngineImpl) DeleteWorkflowExecution(
 
 	// Close workflow execution is deleted using DeleteExecutionTask.
 
-	// DeleteWorkflowExecution is not replicated automatically. Workflow execution must be deleted separately in each cluster.
-	// Although running workflows in active cluster are terminated first and the termination is replicated.
-	// In passive cluster, workflow executions are just deleted in regardless of its status.
+	// DeleteWorkflowExecution is not replicated automatically. Workflow executions must be deleted separately in each cluster.
+	// Although running workflows in active cluster are terminated first and the termination event might be replicated.
+	// In passive cluster, workflow executions are just deleted in regardless of its state.
 
-	if weCtx.GetMutableState().GetExecutionState().GetState() != enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED {
+	if weCtx.GetMutableState().IsWorkflowExecutionRunning() {
```

Comment:
so zombie workflow is treated as if it's a passive workflow?

Reply:
Yes, after checking other code, I came up to this check. Zombie (and other corrupted) executions are just deleted. Active cluster will handle all parent-child relations.